tsdb/engine/tsm1/reader.go

```diff
@@ -796,6 +810,45 @@ func (f *fileAccessor) readBytes(entry *IndexEntry, b []byte) ([]byte, error) {
 	return b[4:n], nil
 }
 
+// getChecksums returns a map[string]bool in the form of blockId: valid (writtenChecksum == calulcatedChecksum)
+func (f *fileAccessor) getChecksums(key string) (map[string]bool, error) {
```

Comment:
I was going to do it that way but I'd end up changing lots of bits and pieces of the code I guess... I like the idea :+1: